### ISSUE TITLE
Some feedback and questions on usability for mod list creators

### ISSUE THREAD

**Issue by Vladabdf** [th3w1zard1/KOTORModSync#40]
I am trying to make a personal mod list to make future installs easy, while also potentially sharing said list with friends. One of the mods involved in this mod list I am trying to make uses HoloPatcher instead of TSLPatcher. While I see there is a dialog drop down for HoloPatcher, I can't find any mention of any users (anywhere!) nor any documentation about the tool on whether or not the usage in a mod list that is being created is exactly the same as TSLPatcher. Is it? If so, it should be stated somewhere, either on the GitHub readme or perhaps even in the readme that comes with the tool download. I do see a brief mention of HoloPatcher in the documentation that simply implies it's better than the TSLPatcher method, but again, no information on usage for mod list creators.

Additionally, in the attached image example, there is the mod in question I am trying to add to my mod list. I am in the `Options` section, and within that section there is an *identical* looking `Instructions` drop down box to the `Instructions` section above the `Options` section. When making a mod list, having all of these drop downs is pretty much mandatory, and navigating the UI without getting lost is a nightmare. To help with readability for the end user, I would heavily recommend using a sub-window pop up instead of boxes within boxes in the tool. Effectively, the `Instructions` drop down within the `Options` section would pop up a window asking for the `Instructions` belonging specifically to that `Option` in question, making navigation far easier.

Lastly, when using the HoloPatcher or TSLPatcher method for mods that use those, specifically those with multiple options, I'm curious, how are you supposed to determine which files/options the patcher in question will use? In the "Luxa Hair Fix" scenario, which is used for the tutorial, each individual option in the mod archive seems to come with its own TSLPatcher + instructions, categorized by folder. What about the [Force Affinity Form Fix + Overhaul](https://deadlystream.com/files/file/2174-force-affinity-form-fix-overhaul-mod/) mod, which uses the HoloPatcher and has the options integrated into the patcher itself, rather than separate folders each with their own patcher .exe?

I also apologize if I appear to be rambling, confusing, or whatever other -ing you wish to come up with (annoying, maybe? Lol). I have been trying to figure this tool out for an embarrassing amount of time, as I thought it would have saved myself a lot of time using one of the Mod Builds with a small handful of additional mods I have used over the years thrown in as well. However, I got frustrated once I realized I would have actually installed them all already and saved myself slightly more headache and a lot more time at the cost of tedium, had I just installed them all the old fashioned way, and now here I am. I am also including my `.toml` in case you would like to look at it.

Any enlightnement on these matters would be wonderful. Thanks.

![2024-06-17_01-46-25 KOTORModSync](https://github.com/th3w1zard1/KOTORModSync/assets/20749232/4e9a9014-9e06-4200-bfd5-d904ad63338a)
[snip.zip](https://github.com/user-attachments/files/15862475/snip.zip)

**Comment by Vladabdf** [comment]
I got some errors and fixed them after some more tinkering. Now I am experiencing a new error that says to report it to a developer. Included are screenshots from it.
![2024-06-17_21-13-04 KOTORModSync](https://github.com/th3w1zard1/KOTORModSync/assets/20749232/dd1226ff-2da5-4a08-aa4e-12ced02dda8f)
![2024-06-17_21-13-15 KOTORModSync](https://github.com/th3w1zard1/KOTORModSync/assets/20749232/0cf384b0-1445-4c6e-880b-0e5fd779c01c)

**Comment by th3w1zard1** [comment]
> I am trying to make a personal mod list to make future installs easy, while also potentially sharing said list with friends.

I'm happy someone is attempting to use this for something other than the main neocities builds. Let's go through your issues one by one.

> One of the mods involved in this mod list I am trying to make uses HoloPatcher instead of TSLPatcher.

This is already default. The global Patcher Option on the righthand side of the app deteremines which patcher will be used for **all mods**. It's now a deprecated option: originally TSLPatcher was the only patcher that this would work with, and holopatcher was newer, so the app used to simply run TSLPatcher. However since many mods are now requiring HoloPatcher, that now causes the TSLPatcher option to be unsupported, since TSLPatcher is not forwards compatible.

> I can't find any mention of any users (anywhere!) nor any documentation about the tool on whether or not the usage in a mod list that is being created is exactly the same as TSLPatcher. Is it? If so, it should be stated somewhere, either on the GitHub readme or perhaps even in the readme that comes with the tool download. I do see a brief mention of HoloPatcher in the documentation that simply implies it's better than the TSLPatcher method, but again, no information on usage for mod list creators.

I'm not sure what you mean. At face value, it sounds like you're trying to ask if our mod lists will output the same results as TSLPatcher. TSLPatcher does not support mod lists. TSLPatcher supports changes.ini. Both holopatcher and tslpatcher install mods written with changes.ini. KOTORModSync allows mod developers to define multi-mod compatibility and installation orders.

tldr: HoloPatcher is *backwards-compatible* with TSLPatcher. TSLPatcher is **not** *forwards-compatible* with HoloPatcher

**Comment by th3w1zard1** [comment]
> Additionally, in the attached image example, there is the mod in question I am trying to add to my mod list. I am in the Options section, and within that section there is an identical looking Instructions drop down box to the Instructions section above the Options section. When making a mod list, having all of these drop downs is pretty much mandatory, and navigating the UI without getting lost is a nightmare. To help with readability for the end user, I would heavily recommend using a sub-window pop up instead of boxes within boxes in the tool. Effectively, the Instructions drop down within the Options section would pop up a window asking for the Instructions belonging specifically to that Option in question, making navigation far easier.

Can agree the UI could be improved. KOTORModSync was quite literally my first app I've written and designed. Many UI things I wish I knew how to do when I was originally creating this.

**Comment by th3w1zard1** [comment]
> Lastly, when using the HoloPatcher or TSLPatcher method for mods that use those, specifically those with multiple options, I'm curious, how are you supposed to determine which files/options the patcher in question will use? In the "Luxa Hair Fix" scenario, which is used for the tutorial, each individual option in the mod archive seems to come with its own TSLPatcher + instructions, categorized by folder. What about the [Force Affinity Form Fix + Overhaul](https://deadlystream.com/files/file/2174-force-affinity-form-fix-overhaul-mod/) mod, which uses the HoloPatcher and has the options integrated into the patcher itself, rather than separate folders each with their own patcher .exe?

This goes back to the first message. KOTORModSync does not **ever** use the patcher that comes with individual mods. It only uses the patcher that comes with the kotormodsync app itself.

**Comment by th3w1zard1** [comment]
> I also apologize if I appear to be rambling, confusing, or whatever other -ing you wish to come up with (annoying, maybe? Lol). I have been trying to figure this tool out for an embarrassing amount of time, as I thought it would have saved myself a lot of time using one of the Mod Builds with a small handful of additional mods I have used over the years thrown in as well. 

Hey no worries. I just wish you asked sooner or I clarified some of these things prior. I struggled putting the documentation together, and since I wrote the tool, I struggled to understand what someone new would be confused by. It's hard to put myself in end user shoes when figuring out the most important things to document.

> However, I got frustrated once I realized I would have actually installed them all already and saved myself slightly more headache and a lot more time at the cost of tedium, had I just installed them all the old fashioned way, and now here I am. I am also including my `.toml` in case you would like to look at it.

In my experience with the 300+ mods this app supports across 6 different lists, it takes about 1.2x longer to define the instructions than it does to install the actual mods separately.  This means if it takes about 2 minutes (120 seconds) to install a mod, it would take 144 seconds to write the same instructions.

The goal here is to make it easier for end users and to absolutely ensure that the start->finish is universally the same.

**Comment by th3w1zard1** [comment]
> I got some errors and fixed them after some more tinkering. Now I am experiencing a new error that says to report it to a developer. Included are screenshots from it. ![2024-06-17_21-13-04 KOTORModSync](https://private-user-images.githubusercontent.com/20749232/340498823-dd1226ff-2da5-4a08-aa4e-12ced02dda8f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzOTI3MDMsIm5iZiI6MTcyMDM5MjQwMywicGF0aCI6Ii8yMDc0OTIzMi8zNDA0OTg4MjMtZGQxMjI2ZmYtMmRhNS00YTA4LWFhNGUtMTJjZWQwMmRkYThmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDIyNDY0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhZWRhMDI0NDFlZGVkNzdhM2IyMDY2OWU2ZWNmN2U0NDUyMGJhYWFmYWFjNmMwMzE0NzUwMjFhMzYyYzBjN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XR9dU6Fu0Gif8SVS6kWSQkQBBnqy0OlYqsBG8N_ogaM) ![2024-06-17_21-13-15 KOTORModSync](https://private-user-images.githubusercontent.com/20749232/340498824-0cf384b0-1445-4c6e-880b-0e5fd779c01c.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzOTI3MDMsIm5iZiI6MTcyMDM5MjQwMywicGF0aCI6Ii8yMDc0OTIzMi8zNDA0OTg4MjQtMGNmMzg0YjAtMTQ0NS00YzZlLTg4MGItMGU1ZmQ3NzljMDFjLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDIyNDY0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxODdmNGE4NjAxMzMxYTE2OTkzNzYyYzJhYzJmZWU5NWE3NGQwODRiMTA1MTdhZDU0ZTViNzhiZmI1MjY5ZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rHJScr5DIRSfn0kP1JeMhvZOP4uGMAUueJdOBDxtfYU)

and finally, this error means that there were no matching files found by your filter. Which means this file isn't found:
```
Source = ["<<modDirectory>>\\ForceAffinityFix\\HoloPatcher.exe"]
```
to fix it i recommend you first set your paths, and then use the `Browse` button to prevent any typos from happening in the path itself.
